services/github/github-workflow-status.service.js

```diff
@@ -1,100 +1,28 @@
-  message: Joi.alternatives()
-    .try(isBuildStatus, Joi.equal('no status'))
-    .required(),
-}).required()
-
-const queryParamSchema = Joi.object({
-  event: Joi.string(),
-}).required()
-
-const keywords = ['action', 'actions']
-
-export default class GithubWorkflowStatus extends BaseSvgScrapingService {
+export default class DeprecatedGithubWorkflowStatus extends BaseService {
   static category = 'build'
 
   static route = {
     base: 'github/workflow/status',
-    pattern: ':user/:repo/:workflow/:branch*',
-    queryParamSchema,
-  }
-
-  static examples = [
-    {
-      title: 'GitHub Workflow Status',
-      pattern: ':user/:repo/:workflow',
-      namedParams: {
-        user: 'actions',
-        repo: 'toolkit',
-        workflow: 'toolkit-unit-tests',
-      },
-      staticPreview: renderBuildStatusBadge({
-        status: 'passing',
-      }),
-      documentation,
-      keywords,
-    },
-    {
-      title: 'GitHub Workflow Status (branch)',
-      pattern: ':user/:repo/:workflow/:branch',
-      namedParams: {
-        user: 'actions',
-        repo: 'toolkit',
-        workflow: 'toolkit-unit-tests',
-        branch: 'master',
-      },
-      staticPreview: renderBuildStatusBadge({
-        status: 'passing',
-      }),
-      documentation,
-      keywords,
-    },
-    {
-      title: 'GitHub Workflow Status (event)',
-      pattern: ':user/:repo/:workflow',
-      namedParams: {
-        user: 'actions',
-        repo: 'toolkit',
-        workflow: 'toolkit-unit-tests',
-      },
-      queryParams: {
-        event: 'push',
-      },
-      staticPreview: renderBuildStatusBadge({
-        status: 'passing',
-      }),
-      documentation,
-      keywords,
-    },
-  ]
-
-  static defaultBadgeData = {
-    label: 'build',
-  }
-
-  async fetch({ user, repo, workflow, branch, event }) {
-    const { message: status } = await this._requestSvg({
-      schema,
-      url: `https://github.com/${user}/${repo}/workflows/${encodeURIComponent(
-        workflow
-      )}/badge.svg`,
-      options: { searchParams: { branch, event } },
-      valueMatcher: />([^<>]+)<\/tspan><\/text><\/g><path/,
-      errorMessages: {
-        404: 'repo, branch, or workflow not found',
-      },
-    })
-
-    return { status }
+    pattern: ':various+',
   }
 
-  async handle({ user, repo, workflow, branch }, { event }) {
-    const { status } = await this.fetch({ user, repo, workflow, branch, event })
-    return renderBuildStatusBadge({ status })
+  static examples = []
+
+  static defaultBadgeData = { label: 'build' }
+
+  async handle() {
+    return {
+      label: 'build',
+      message: 'https://github.com/badges/shields/issues/8671',
+      /*
+      This is a 'special' deprecation because we are making a breaking change
+      We've implemented it as a custom class instead of a normal
+      deprecatedService so that we can include link.
+      */
+      link: ['https://github.com/badges/shields/issues/8671'],
```

Comment:
Because `deprecatedService()` throws an exception, we can't customise it beyond the label/message so I've done this as a custom class. I think the number of cases where this will be actually clickable will be low but it is worth doing IMO. This is a sufficiently exceptional edge-case that I don't mind this being a bit special and wierd and custom.